### PR TITLE
Add CheckServiceExistence Rule for gke-ingress-checker

### PR DESCRIPTION
**Pull request by songrx1997**
Rename util package in gke-diagnoser to report
Add CheckServiceExistence rule for gke-ingress-checker
Add unit test for CheckServiceExistence rule

### CODE REVIEW

**Comment by bowei** [comment]
I thought we were going to rename the cmd name? The directory is still named gke-diagnoser. Let's not forget to fix this.

File: cmd/gke-diagnoser/app/ingress/rule.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+func CheckServiceExistence(ns string, svcName string, client clientset.Interface) (*corev1.Service, string, string) {
+	svc, err := client.CoreV1().Services(ns).Get(context.TODO(), svcName, metav1.GetOptions{})
+	if err != nil {
```
Comment (bowei) [inline cmd/gke-diagnoser/app/ingress/rule.go:30]: You need to check that the error is actually a not found error.
If there was a different kind of error, you need to surface that to the user and also print a different message.

Comment (songrx1997) [reply cmd/gke-diagnoser/app/ingress/rule.go:31]: Updated, thanks!

**Comment by songrx1997** [comment]
> I thought we were going to rename the cmd name? The directory is still named gke-diagnoser. Let's not forget to fix this.

Will rename the directory after current PRs merged

File: cmd/gke-diagnoser/app/ingress/rule_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+			name:      "service-1",
+			expect:    "PASSED",
+		},
+		{
+			desc:      "correct namespace and wrong name",
+			namespace: "namespace1",
+			name:      "service-2",
+			expect:    "FAILED",
+		},
+		{
```
Comment (swetharepakula) [inline cmd/gke-diagnoser/app/ingress/rule_test.go:60]: duplicate testcase

Comment (songrx1997) [reply cmd/gke-diagnoser/app/ingress/rule_test.go:60]: Removed, thanks!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve

/hold cancel
The rename will be handled in a separate PR

**Pull request merged**